k8s-operator/sessionrecording/ws/conn.go

```diff
@@ -28,14 +28,16 @@ import (
 // The hijacked connection is used to transmit *.channel.k8s.io streams between Kubernetes client ('kubectl') and the destination proxy controlled by Kubernetes.
 // Data read from the underlying network connection is data sent via one of the streams from the client to the container.
 // Data written to the underlying connection is data sent from the container to the client.
-// We parse the data and send everything for the STDOUT/STDERR streams to the configured tsrecorder as an asciinema recording with the provided header.
+// We parse the data and send everything for the stdout/stderr streams to the configured tsrecorder as an asciinema recording with the provided header.
 // https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/4006-transition-spdy-to-websockets#proposal-new-remotecommand-sub-protocol-version---v5channelk8sio
-func New(c net.Conn, rec *tsrecorder.Client, ch sessionrecording.CastHeader, log *zap.SugaredLogger) net.Conn {
+func New(c net.Conn, rec *tsrecorder.Client, ch sessionrecording.CastHeader, hasTerm bool, log *zap.SugaredLogger) net.Conn {
 	return &conn{
-		Conn: c,
-		rec:  rec,
-		ch:   ch,
-		log:  log,
+		Conn:               c,
+		rec:                rec,
+		ch:                 ch,
+		hasTerm:            hasTerm,
+		log:                log,
+		initialTermSizeSet: make(chan string, 1),
```

Comment:
```suggestion
		initialTermSizeSet: make(chan struct{}),
```

Note: This suggestion needs to be paired with the suggestion about using `close` on the channel for it to work properly.